crates/ruff_linter/src/rules/flake8_pytest_style/rules/raises.rs

```diff
@@ -151,6 +151,52 @@ impl Violation for PytestRaisesWithoutException {
     }
 }
 
+/// ## What it does
+/// Checks for non-raw literal string arguments passed to the `match` parameter
+/// of `pytest.raises()` that does not contain any regex metacharacters.
```

Comment:
This is different to the rule I suggested in https://github.com/astral-sh/ruff/issues/13705#issuecomment-2532259079. I suggested having a rule that only emits a diagnostic if the string _contains_ a metacharacter. But your implementation only emits a diagnostic if the string _does not contain_ a metacharacter.

The reason I think it would be better to emit the error if the string _contains_ a metacharacter is that I think it's quite a common footgun to do things like this:

```py
import pytest

def do_thing_that_raises():
    raise Exception('foobar')

def test_error_message_contains_foo_at_end_of_sentence():
    with pytest.raises(Exception, match="foo."):
        do_thing_that_raises()
```

Where that `.` will match any character at all (since the string is interpreted as a regex, and `.` is a regex metacharacter). So the test would erroneously pass, when it should have failed. I think it's reasonable to say that the user should either make it explicit that they intend the character to be a regex metacharacter (by using a raw string -- `r"foo."`), or double-escape the metacharacter (`"foo\\."`), or wrap the call in `re.escape()` (`re.escape("foo.")`).

I think if you implement the rule I suggested, it will be much less noisy